Sources/BraintreePayPal/BTPayPalAnalytics.swift

```diff
@@ -0,0 +1,21 @@
+import Foundation
+
+class BTPayPalAnalytics {
+    static let vaultRequestStarted = "paypal:vault-tokenize:started"
+    static let checkoutRequestStarted = "paypal:checkout-tokenize:started"
+    
+    // counted in conversion rates
+    static let tokenizeFailed = "paypal:tokenize:failed"
+    static let tokenizeSucceeded = "paypal:tokenize:succeeded"
+    static let browserLoginCanceled = "paypal:tokenize:browser-login:canceled"
+   
+    // additional detail analytic messages
+    static let tokenizeNetworkConnectionFailed = "paypal:tokenize:network-connection:failed"
+    static let tokenizeBrowserSwitchNetworkConnectionLost = "paypal:tokenize:browser-switch-network-connection:failed"
+
+    static let browserPresentationStarted = "paypal:tokenize:browser-presentation:started"
+    static let browserPresentationSucceeded = "paypal:tokenize:browser-presentation:succeeded"
+    static let browserPresentationFailed = "paypal:tokenize:browser-presentation:failed"
+    static let authsessionBrowserCancel = "paypal:tokenize:authsession-browser:canceled"
+    static let authsessionAlertCancel = "paypal:tokenize:authsession-alert:canceled"
```

Comment:
I discussed this with Sammy and she thought that this distinction of where it was cancel might be meaningful but I think weren't 100% sure.

Reply:
I think similar to the "fail" case, we want to send a "generic cancel" for both cancel types (the "Done" on the ASWeb, and via the Alert). This was we can easily calculate cancelation rates without needing to add events up.

However, I do think it will be useful to have an "additional detail event" for the alert cancel case. Only b/c recently Starbucks & Uber were wanting our stats on it, and we had none that were accurate.

I don't think we need an "additional detail event" (aka non conversion one) for the "Done" button pressed case.

What do y'all think?

Reply:
This is technically another thing we could differentiate in some metadata value on the generic event. Do we want to explore that option?

Something like:
```
{
  "event_name": "paypal:tokenize:canceled", 
    "metadata": {
        "client_os": "iOS 15.2",
        ...
        "event_details": "alert_canceled" // or "browser_canceled"
    }
}

```

Reply:
There is the case we discussed where user clicks on the hyperlink to cancel and that rn, just sends tokenizeFailed and BTPayPalError.canceled. Should I make another specific cancel message for this or do we just care about the cancels from the buttons?

Reply:
The only specific cancel that I think we care about is the one from the alert. The others can all just be generic cancel cases.